src/book/03-guides/01-main-codebase/07-onboarding-task.mdx

```diff
@@ -0,0 +1,486 @@
+
+**Requirements**
+
+1. The code must be written in C++, using NUClear.
+
+2. The code must be compiled and installed into a robot.
+
+3. Once the right answer has been derived, one must emit that answer. If the correct answer was emitted, the robot should nod yes.
+
+4. You must use at least 3 modules to implement their solutions.
+
+   - Why at least 3 modules? One will start the reaction, one will determine when to stop sending the message to the other reactor and start emitting the final answer, and one will process the final answer and send the NodYes command.
+   - Decoupling is very important for code reliability and robustness. It also helps to make your code easy to review.
+
+5. You must make your own branch and submit your code as a pull request for review once finished.
+
+**Constraints**
+
+1. You cannot use a `for` loop or a `while` loop to evaluate the expression.
+
+2. You cannot use the proposed $n(n+1)/2$ shortcut to evaluate the expression
+
+3. Only `on<Trigger<...>>` and `emit()` statements can be used.
+
+**Recommendations**
+
+- Use `./b module generate <absolute module path and name>` to generate your modules. A full path example could be something like `test/onboarding_task`.
+
+- Use the NUClear role from the start for this task and a starting point. It looks like
+
+  ```make
+  nuclear_role(
+  extension::FileWatcher
+  support::logging::ConsoleLogHandler
+  actuation::KinematicsConfiguration
+  platform::${SUBCONTROLLER}::HardwareIO
+  input::SensorFilter
+  extension::Director
+  actuation::Kinematics
+  actuation::Servos
+  onboarding::Ping
+  onboarding::Pong
+  onboarding::Judge
+  )
+  ```
+
+- They say that naming things is one of the hardest tasks in software engineering. What should the names of your modules be? Should they still be Ping and Pong? Can you think of more suitable names? What about your the names of your messages, i.e., your `.proto` files?
+
+- NUbook will have all the information you need, and then some. This task cannot explain NUClear in its full glory, but it should be enough to enable members to contribute to the code base as soon as possible. If you wish to know more, you must learn to read documentation and if you are still stuck, don't hesitate to ask questions.
+
+<Alert type='info'>
+
+When in VS code, press <kbd>CTRL</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd> to find code snippets. This is particularly useful if members want to know how a specific bit of code should be used or how it is implemented. This will be specifically useful when looking at where to put your files in. For example, proto messages should lie in `~/NUbots/shared/messages/...` and modules should lie within `~/NUbots/modules`.
```

Comment:
```suggestion
When in VS Code, press <kbd>CTRL</kbd> + <kbd>Shift</kbd> + <kbd>F</kbd> to find code snippets. This is particularly useful if members want to know how a specific bit of code should be used or how it is implemented. This will be specifically useful when looking at where to put your files in. For example, proto messages should lie in `shared/messages/...` and modules should lie within `modules`.
```
maybe not having absolute paths as people may not be cloning to their home folder